### PR TITLE
[TraceDecoder] Add decrypted data

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

The options `trace_decode 1` does not log the decrypted payload.

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

File: examples/common/tracing/TraceDecoder.cpp [hunk 1]
```diff
@@ -272,6 +272,13 @@ CHIP_ERROR TraceDecoder::MaybeLogAndConsumePayload(Json::Value & json, bool isRe
     auto size = static_cast<uint16_t>(json[kPayloadSizeKey].asLargestUInt());
     if (size)
     {
+        {
+            auto scoppedIndent = ScopedLogIndentWithSize("Decrypted Payload", size);
```
Comment (bzbarsky-apple) [inline examples/common/tracing/TraceDecoder.cpp:276]:
```suggestion
            auto scopedIndent = ScopedLogIndentWithSize("Decrypted Payload", size);
```

File: examples/common/tracing/TraceDecoder.cpp [hunk 2]
```diff
@@ -272,6 +272,13 @@ CHIP_ERROR TraceDecoder::MaybeLogAndConsumePayload(Json::Value & json, bool isRe
...
+
+            auto payload = json[kPayloadDataKey].asString();
+            Log("data", payload.c_str());
```
Comment (bzbarsky-apple) [inline examples/common/tracing/TraceDecoder.cpp:279]: Payload can have embedded nulls, no?  We need to do whatever we would normally do to log a random bytespan here...

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/common/tracing/TraceDecoder.cpp (hunk 2 above)
Comment (tcarmelveilleux) [reply examples/common/tracing/TraceDecoder.cpp:279]: No, the payload at this layer is hex

Comment (vivien-apple) [reply examples/common/tracing/TraceDecoder.cpp:278]: Ah @tcarmelveilleux was faster than me.
Basically `kPayloadDataKey` maps to `payload_hex` which is encoded as hex there:
 *  https://github.com/project-chip/connectedhomeip/blob/caacbdef39b2ce69a918979cefd2c5d8448a22e1/examples/common/tracing/TraceHandlers.cpp#L282
 * https://github.com/project-chip/connectedhomeip/blob/caacbdef39b2ce69a918979cefd2c5d8448a22e1/examples/common/tracing/TraceHandlers.cpp#L236
 * https://github.com/project-chip/connectedhomeip/blob/caacbdef39b2ce69a918979cefd2c5d8448a22e1/examples/common/tracing/TraceHandlers.cpp#L258

Comment (bzbarsky-apple) [reply examples/common/tracing/TraceDecoder.cpp:278]: Ah, ok, great.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**